Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,552 @@
+
+        if (levelEntityId.IsValid())
+        {
+            AZ::Entity* levelEntity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(levelEntity, &AZ::ComponentApplicationBus::Events::FindEntity, levelEntityId);
+            if (levelEntity)
+            {
+                LocalViewBookmarkComponent* bookmarkComponent = levelEntity->FindComponent<LocalViewBookmarkComponent>();
+                if (bookmarkComponent)
+                {
+                    return bookmarkComponent;
+                }
+                //If we didn't find a component then we add it and return it.
+                levelEntity->Deactivate();
+                levelEntity->CreateComponent<LocalViewBookmarkComponent>();
+                levelEntity->Activate();
+                bookmarkComponent = levelEntity->FindComponent<LocalViewBookmarkComponent>();
+
+                if (bookmarkComponent)
+                {
+                    return bookmarkComponent;
+                }
+            }
+        }
+        return nullptr;
+    }
+
+    AZStd::string ViewBookmarkLoader::GenerateBookmarkFileName() const
+    {
+        auto* prefabEditorEntityOwnershipInterface = AZ::Interface<AzToolsFramework::PrefabEditorEntityOwnershipInterface>::Get();
+        AZ_Assert(prefabEditorEntityOwnershipInterface != nullptr, "PrefabEditorEntityOwnershipInterface is not found.");
+        AzToolsFramework::Prefab::TemplateId rootPrefabTemplateId = prefabEditorEntityOwnershipInterface->GetRootPrefabTemplateId();
+
+        if (rootPrefabTemplateId != AzToolsFramework::Prefab::InvalidTemplateId)
+        {
+            auto* prefabSystemComponent = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
+            AZ_Assert(
+                prefabSystemComponent != nullptr,
+                "Prefab System Component Interface could not be found. "
+                "It is a requirement for the ViewBookmarkLoader class. "
+                "Check that it is being correctly initialized.");
+            auto prefabTemplate = prefabSystemComponent->FindTemplate(rootPrefabTemplateId);
+            AZStd::string prefabTemplateName = prefabTemplate->get().GetFilePath().Filename().Stem().Native();
+            // To generate the file name in which we will store the view Bookmarks we use the name of the prefab + the timestamp
+            // e.g. LevelName_1639763579377.setreg
+            AZStd::chrono::system_clock::time_point now = AZStd::chrono::system_clock::now();
+            AZStd::string sTime = AZStd::string::format("%llu", now.time_since_epoch().count());
+
+            return prefabTemplateName.data() + AZStd::string("_") + sTime + ".setreg";
+        }
+        return AZStd::string();
+    }
+
+    bool ViewBookmarkLoader::SaveSharedBookmark([[maybe_unused]]ViewBookmark& bookmark)
+    {
+        AZ_Assert(false, "Shared Bookmark Component functionality not implemented.");
+        return false;
+    }
```

Comment:
update: Similar to earlier point, it might be best to just remove _all_ references to SharedBookmarks from this part of the API for now